### PR TITLE
Fixes menu problem with long shop names

### LINKED ISSUE #3967: New header displaying badly in Spanish, Catalan & Portuguese
## Description
When testing the new header on Katuma, I noticed it is displaying badly for certain language. Basically all language except Italian.

## Steps to Reproduce
1. Go to Katuma's staging
2. Go to a shop
3. Switch between italian and something else

## Animated Gif/Screenshot
Italian:
image

Catalan:

image

## Your Environment

* Version used: v2.1
* Browser name and version: Firefox 67
* Operating System and version (desktop or mobile): Desktop Windows 10

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #3967

We make the menu items compact for screens below 1600px (it was 1450px previously).
We should also make the mobile menu appear for screens below 1200px (it is around 1000px now) **but I can't find where this is done**. Any idea?

#### What should we test?
Issue will not happen and menu displays correctly for different screen widths with different data setups:
- logged in, not logged in
- shopping somewhere, not shopping anywhere
- different languages

#### Release notes
Changelog Category: Fixed
Fixed display problem with menu: the menu is not breaking in two lines now in some situations.

### CODE REVIEW

File: app/assets/stylesheets/darkswarm/menu.css.scss [hunk 1]
```diff
@@ -87,7 +87,7 @@ nav.top-bar {
       border-left: 1px solid #ddd;
       padding: 0 14px;
 
-      @media screen and (max-width: 1450px) {
+      @media screen and (max-width: 1600px) {
```
Comment (Matt-Yorkley) [inline app/assets/stylesheets/darkswarm/menu.css.scss]: Do these breakpoint width values appear in any other css files?

**Comment by Matt-Yorkley** [comment]
Turns out we do set a character limit:

limit

**Comment by kristinalim** [comment]
Do you think it's acceptable to do these in _medium_-sized screens? (Actually, it is still "large" in Foundation vocabulary.) Tagging @RachL here.

1. Hide the text for language and "Profile" in the top bar, so that what's left would be the globe and the user icons. Maybe we shouldn't do it for the "Log In" menu item - some visitors might not understand the padlock and would not bother to find a way to log in.
    
    This reminds me of Yuko's design for CERES, where the Login button is spelled out. Maybe the reason for that is similar.
2. Hide "Shopping @".
3. We set a max width for the shop name in the top bar. Beyond that width, we truncate the text and show an ellipsis.

@luisramos0 I recommend against increasing the breakpoint for showing the off-canvas (hamburger) menu as a workaround for this (if really planned, then it's okay). :slightly_smiling_face: We might start considering all of &lt;1200px as "mobile" devices, but some/many of these might prefer the desktop experience.

**Comment by kristinalim** [comment]
Please see here:

20190703023700-top_bar_compressed

@luisramos0 I tested with real changes to see what could work, sorry. I have the changes in this branch: https://github.com/kristinalim/openfoodnetwork/commits/fix/3967-fix_menu_issues

**Comment by luisramos0** [comment]
@kristinalim  amazing! 👏
I think that's awesome, maybe point 1 and 3 are enough and we can keep Shopping@ all the time?

I included your commits in my branch, you can push directly to my branch ;-)

Code climate is complaining about the underscores in the classname.

**Comment by kristinalim** [comment]
@luisramos0 Ah yes! I changed the underscores to hyphens for now. I don't know yet how we should reconfigure `scss-lint` or if I should stop using that CSS class naming convention.

With the "Shopping @", there was still some relatively high risk of content exceeding the allowed area, so I reduced the padding between menu items. This is ready for review again.

**Comment by luisramos0** [comment]
awesome!!!!

**Comment by luisramos0** [comment]
well, none of the commits is mine now so, although GH doesnt let me review the PR, I approve it :-)
moving to test ready.

**Comment by RachL** [comment]
Posting here directly because images in google doc are not displaying correctly.
Windows 7 Firefox 66 not logged in:
image

Windows 7 Firefox 66 logged in and shopping:
image

Windows 7 Firefox 66 Italiano (but other languages were tested):
image

Windows 7 Firefox 66 logged in and not shopping:
image

Windows 10 Edge 18 not logged in:
image

Windows 10 Edge 18 logged in and shopping:
image

Windows 10 Edge 18 Italiano (but other languages were tested):
image

Windows 10 Edge 18 logged in and not shopping:
image

Android 9 Galaxy s10 not logged in:
image

Android 9 Galaxy s10 logged in and shopping:
image

Other language OK on mobile

My browser crashes loading the VM for iphone, will have to sort this out, but I think this PR is ready to go! awesome team work here :muscle: 

One last comment: I'm being picky here, but next time we are working on UI, it would be great to take the time to add the grey line Yuko has designed to mark the separation between the header and the rest of the page. When you scroll on a white background the header is falling :D See here:

image

@Matt-Yorkley @kristinalim @luisramos0  or anyone else... is it just me or it could be interesting to open an issue for this?

**Comment by luisramos0** [comment]
@RachL maybe the request for that line is already in some "mobile ready" issue?

**Comment by RachL** [comment]
@luisramos0 it looks like the main header won't be touched again in future mobile work, but maybe I'm wrong? Or we squease it in https://github.com/openfoodfoundation/openfoodnetwork/issues/2714 ?

**Comment by luisramos0** [comment]
yes, could be, @kirstenalarsen will know better.

**Comment by mkllnk** [comment]
@RachL I'm missing that separation between header and page, too! It would be a good first issue.

**Pull request merged**